### PR TITLE
extensive smart wallet tests & AssetToken whitelist bugfix 

### PR DESCRIPTION
**Pull request by ungaro**
## What's new in this PR?

Coverage for smart wallet contracts increased a lot.

Also there was a bug that prevented AssetToken to be deployed with whitelistenabled flag as the owner was not getting set in constructor.
379326793-d9acd5b5-6f34-4ec9-bea7-2b9c81e87b4c

### CODE REVIEW

**Comment by ungaro** [comment]
`forge coverage --ir-minimum` works now, and master is merged. 
Screenshot 2024-12-18 at 1 25 14 PM

**Comment by ungaro** [comment]
Screenshot 2024-12-18 at 10 55 31 PM

File: smart-wallets/src/extensions/AssetVault.sol [hunk 1]
```diff
@@ -304,6 +313,8 @@ contract AssetVault is WalletUtils, IAssetVault {
      * @param amount Amount of AssetTokens included in this yield allowance
      * @param expiration Timestamp at which the yield expires
      */
+
+    /*
```
Comment (eyqs) [inline smart-wallets/src/extensions/AssetVault.sol]: Remove extra function

**Review by eyqs — APPROVED** [review]
LGTM

**Pull request merged**